### PR TITLE
ALPN updates for mac. Disabled some tests that thanks to Apple, will never pass until we implement our own dns resolvers.

### PR DESCRIPTION
**Pull request by JonathanHenson**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/darwin/secure_transport_tls_channel_handler.c [hunk 1]
```diff
@@ -12,23 +12,29 @@
...
-#include <aws/io/pki_utils.h>
+#include <dlfcn.h>
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-variable"
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #pragma clang diagnostic ignored "-Wunused-function"
 
+static OSStatus (*s_SSLSetALPNProtocols_fn_ptr)(SSLContextRef context, CFArrayRef protocols) = NULL;
+static OSStatus (*s_SSLCopyALPNProtocols_fn_ptr)(SSLContextRef context, CFArrayRef *protocols) = NULL;
```
Comment (justinboswell) [inline source/darwin/secure_transport_tls_channel_handler.c]: [trivial] the _fn_ptr seems redundant, maybe just _fn if you really want?

File: source/darwin/secure_transport_tls_channel_handler.c [hunk 2]
```diff
@@ -462,8 +499,11 @@ static int s_process_read_message(
         }
     }
 
+    size_t downstream_window = SIZE_MAX;
     /* process as much as we have queued that will fit in the downstream window. */
-    size_t downstream_window = aws_channel_slot_downstream_read_window(slot);
+    if (slot->adj_right) {
```
Comment (justinboswell) [inline source/darwin/secure_transport_tls_channel_handler.c]: Every time I read this, I wish it was labeled downstream and upstream :(

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -67,9 +67,11 @@ add_test_case(channel_cancels_pending_tasks)
 
 if (NOT WIN32)
     add_test_case(channel_message_passing)
-    add_test_case(test_default_with_ipv6_lookup)
-    add_test_case(test_resolver_ipv6_address_lookup)
-    add_test_case(test_default_with_multiple_lookups)
+    if (NOT APPLE)
```
Comment (justinboswell) [inline tests/CMakeLists.txt]: Probably worth a comment here to explain why not apple

File: source/darwin/secure_transport_tls_channel_handler.c (hunk 1 above)
Comment (JonathanHenson) [reply source/darwin/secure_transport_tls_channel_handler.c]: sure

File: tests/CMakeLists.txt (hunk 3 above)
Comment (JonathanHenson) [reply tests/CMakeLists.txt]: ok

File: source/darwin/secure_transport_tls_channel_handler.c (hunk 2 above)
Comment (JonathanHenson) [reply source/darwin/secure_transport_tls_channel_handler.c]: ¯\_(ツ)_/¯

**Pull request merged**